flocker/provision/test/test_install.py

```diff
@@ -411,3 +415,69 @@ def test_with_branch_and_version(self):
             run(command="yum install --enablerepo=clusterhq-build "
                         "-y clusterhq-flocker-node-1.2.3-1")
         ]))
+
+
+class PrivateKeyLoggingTest(SynchronousTestCase):
+
+    def test_private_key_removed(self):
+        """
+        A private key is removed for logging.
+        """
+        key = dedent('''
+            -----BEGIN PRIVATE KEY-----
+            MFDkDKSLDDSf
+            MFSENSITIVED
+            MDKODSFJOEWe
+            -----END PRIVATE KEY-----
+            ''')
+        self.assertIn('SENSITIVE', key)
+        logged = _remove_private_key(key)
+        self.assertIn('-----BEGIN PRIVATE KEY-----', logged)
+        self.assertNotIn('SENSITIVE', logged)
+        self.assertIn('-----END PRIVATE KEY-----', logged)
+
+    def test_non_key_kept(self):
+        """
+        Non-key data is kept for logging.
+        """
+        key = 'some random data, not a key'
+        self.assertEqual(key, _remove_private_key(key))
+
+    def test_short_key_kept(self):
+        """
+        A key that is suspiciously short is kept for logging.
+        """
+        key = dedent('''
+            -----BEGIN PRIVATE KEY-----
+            short
+            -----END PRIVATE KEY-----
+            ''')
+        self.assertEqual(key, _remove_private_key(key))
+
+    def test_no_end_key_removed(self):
+        """
+        A missing end tag doe snot prevent removal working.
+        """
+        key = dedent('''
+            -----BEGIN PRIVATE KEY-----
+            MFDkDKSLDDSf
+            MFSENSITIVED
+            MDKODSFJOEWe
+            ''')
+        self.assertIn('SENSITIVE', key)
+        self.assertNotIn('SENSITIVE', _remove_private_key(key))
+
+
+class DatasetLoggingTest(SynchronousTestCase):
+
+    def test_dataset_logged_safely(self):
+        config = {
+            'dataset': {
+                'secret': 'SENSITIVE',
+                'zone': 'keep'
+                }
+            }
+        content = yaml.safe_dump(config)
+        logged = _remove_dataset_fields(content)
+        self.assertNotIn(logged, 'SENSITIVE')
```

Comment:
A better test without only one assertion would probably just be `self.assertEqual(yaml.safe_load(logged), {'secret': 'REMOVED', 'zone': 'keep'})`